### DISCUSSION TITLE
Facing issue in prisma when using orderBy and cursor together in count method.

### DISCUSSION THREAD

**Discussion by jaybhattjb** [prisma/prisma#13348] [Q&A]
Hi Team, 
My requirement is to get the count for the model based on where, orderBy, and cursor as below : 
findCountArgs : 
`
{
  "where": {
    "userId": 80,
    "isDeleted": false,
    "contentType": {
      "in": [
        "Image"
      ]
    }
  },
  "orderBy": [
    {
      "isPinned": "desc"
    },
    {
      "pinnedDate": "desc"
    },
    {
      "createdAt": "desc"
    }
  ],
  "cursor": {
    "id": 401
  }
}
`

`nextCount = await prismaModel.count({
      where: Object.assign({}, where),
      cursor: { id: cursor },
      orderBy
});`

But facing the below error when using cursor and orderBy together. 

Error : 
'Input error. Unable to process combination of query arguments for aggregation query. Please note that it is not possible at the moment to have a null-cursor, or a cursor and orderBy combination that not stable (unique).'

Can you please check and revert ASAP?

**Comment by richardwardza** [comment]
Does your `orderBy` contain any fields that are not marked as `unique` in your schema?

My reading of the error message is that if you use `cursor` and `orderBy` then your fields in the `orderBy` need to all be unique in your schema.

**Comment by jaybhattjb** [comment]
Yes, in "orderBy" using createdAt and a few other fields which are not unique.

**Reply by richardwardza** [reply]
That's where the issue currently is then. From the error message Prisma doesn't currently support that "Please note that it is not possible at the moment to have a cursor and orderBy combination that not stable (unique)".

It seems that for now, if you make the fields unique, then it should work..

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
